mkPersist seems to be incompatible with latest persistent classes

Trying out stackage nightly-2020-02-06 I get the following errors with my call to mkPersist:
```
/home/elodie/clients/dolla/ico-board/app/Ico/Types.hs:35:1: error:
    ‘requireUniquesP’ is not a (visible) method of class ‘persistent-2.10.5.1:Database.Persist.Class.PersistUnique.AtLeastOneUniqueKey’
   |
35 | share [ mkPersist sqlSettings
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...

/home/elodie/clients/dolla/ico-board/app/Ico/Types.hs:35:1: error:
    ‘onlyUniqueP’ is not a (visible) method of class ‘persistent-2.10.5.1:Database.Persist.Class.PersistUnique.OnlyOneUniqueKey’
   |
35 | share [ mkPersist sqlSettings
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
```
Looking at the code for mkPersist just a little, it looks like the problem is related to the definition for `mkUniqueKeyInstances` in persistent-template/src/Database/Persist/TH.hs.

My system details can be found in this gist: https://gist.github.com/nlander/3202bbd001006b9c1197e23b77ca9047

That fix worked for me!

I do get a new warning though:
```
/home/elodie/code/persistent-compile-error-demo/app/Main.hs:28:1: warning: [-Wmissing-deriving-strategies]
    No deriving strategy specified. Did you want stock, newtype, or anyclass?
   |
28 | share [ mkPersist sqlSettings
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
```

--- a/persistent-template/ChangeLog.md
+++ b/persistent-template/ChangeLog.md
@@ -1,5 +1,9 @@
 ## Unreleased changes
 
+## 2.8.2.2
+
+* Fix the `mkPersist` function to not require importing the classes explicitly. Also add more of the required extensions to the error message. []()
+
 ## 2.8.2.1
 
 * Fix the test-suite for persistent-template. [#1023](https://github.com/yesodweb/persistent/pull/1023)

--- a/persistent-template/ChangeLog.md
+++ b/persistent-template/ChangeLog.md
@@ -2,7 +2,7 @@
 
 ## 2.8.2.2
 
-* Fix the `mkPersist` function to not require importing the classes explicitly. Also add more of the required extensions to the error message. []()
+* Fix the `mkPersist` function to not require importing the classes explicitly. Also add more of the required extensions to the error message. [#1027](https://github.com/yesodweb/persistent/pull/1027)
 
 ## 2.8.2.1
 

--- a/persistent-template/ChangeLog.md
+++ b/persistent-template/ChangeLog.md
@@ -2,7 +2,7 @@
 
 ## 2.8.2.2
 
-* Fix the `mkPersist` function to not require importing the classes explicitly. Also add more of the required extensions to the error message. [#1027](https://github.com/yesodweb/persistent/pull/1027)
+* Fix the `mkPersist` function to not require importing the classes explicitly. [#1027](https://github.com/yesodweb/persistent/pull/1027)
 
 ## 2.8.2.1
 

--- a/persistent-template/Database/Persist/TH.hs
+++ b/persistent-template/Database/Persist/TH.hs
@@ -399,7 +399,6 @@ mkEntityDefSqlTypeExp emEntities entityMap ent =
 -- 'EntityDef's. Works well with the persist quasi-quoter.
 mkPersist :: MkPersistSettings -> [EntityDef] -> Q [Dec]
 mkPersist mps ents' = do
-    requirePersistentExtensions
     x <- fmap Data.Monoid.mconcat $ mapM (persistFieldFromEntity mps) ents
     y <- fmap mconcat $ mapM (mkEntity entityMap mps) ents
     z <- fmap mconcat $ mapM (mkJSON mps) ents
@@ -1877,9 +1876,5 @@ requirePersistentExtensions = do
         , GeneralizedNewtypeDeriving
         , StandaloneDeriving
         , UndecidableInstances
-        , TypeFamilies
-        , MultiParamTypeClasses
-        , FlexibleInstances
-        , GADTs
         ]
     extensionToPragma ext = "{-# LANGUAGE " <> show ext <> " #-}"

--- a/persistent-template/Database/Persist/TH.hs
+++ b/persistent-template/Database/Persist/TH.hs
@@ -399,6 +399,7 @@ mkEntityDefSqlTypeExp emEntities entityMap ent =
 -- 'EntityDef's. Works well with the persist quasi-quoter.
 mkPersist :: MkPersistSettings -> [EntityDef] -> Q [Dec]
 mkPersist mps ents' = do
+    requirePersistentExtensions
     x <- fmap Data.Monoid.mconcat $ mapM (persistFieldFromEntity mps) ents
     y <- fmap mconcat $ mapM (mkEntity entityMap mps) ents
     z <- fmap mconcat $ mapM (mkJSON mps) ents
@@ -1128,8 +1129,8 @@ mkUniqueKeyInstances mps t = do
         [_] -> mappend <$> singleUniqueKey <*> atLeastOneKey
         (_:_) -> mappend <$> typeErrorMultiple <*> atLeastOneKey
   where
-    requireUniquesPName = mkName "requireUniquesP"
-    onlyUniquePName = mkName "onlyUniqueP"
+    requireUniquesPName = 'requireUniquesP
+    onlyUniquePName = 'onlyUniqueP
     typeErrorSingle = mkOnlyUniqueError typeErrorNoneCtx
     typeErrorMultiple = mkOnlyUniqueError typeErrorMultipleCtx
 
@@ -1160,7 +1161,7 @@ mkUniqueKeyInstances mps t = do
             [ Clause
                 [ WildP ]
                 (NormalB
-                    (VarE (mkName "error") `AppE` LitE (StringL "impossible"))
+                    (VarE 'error `AppE` LitE (StringL "impossible"))
                 )
                 []
             ]
@@ -1871,5 +1872,14 @@ requirePersistentExtensions = do
                     ]
 
   where
-    requiredExtensions = [DerivingStrategies, GeneralizedNewtypeDeriving, StandaloneDeriving, UndecidableInstances]
+    requiredExtensions =
+        [ DerivingStrategies
+        , GeneralizedNewtypeDeriving
+        , StandaloneDeriving
+        , UndecidableInstances
+        , TypeFamilies
+        , GADTs
+        , MultiParamTypeClasses
+        , FlexibleInstances
+        ]
     extensionToPragma ext = "{-# LANGUAGE " <> show ext <> " #-}"

--- a/persistent-template/Database/Persist/TH.hs
+++ b/persistent-template/Database/Persist/TH.hs
@@ -1878,8 +1878,8 @@ requirePersistentExtensions = do
         , StandaloneDeriving
         , UndecidableInstances
         , TypeFamilies
-        , GADTs
         , MultiParamTypeClasses
         , FlexibleInstances
+        , GADTs
         ]
     extensionToPragma ext = "{-# LANGUAGE " <> show ext <> " #-}"

--- a/persistent-template/persistent-template.cabal
+++ b/persistent-template/persistent-template.cabal
@@ -1,5 +1,5 @@
 name:            persistent-template
-version:         2.8.2.1
+version:         2.8.2.2
 license:         MIT
 license-file:    LICENSE
 author:          Michael Snoyman <michael@snoyman.com>

--- a/persistent-test/persistent-test.cabal
+++ b/persistent-test/persistent-test.cabal
@@ -35,6 +35,7 @@ library
                      MpsNoPrefixTest
                      PersistentTest
                      PersistentTestModels
+                     PersistentTestModelsImports
                      PersistTestPetType
                      PersistTestPetCollarType
                      PersistUniqueTest

--- a/persistent-test/src/PersistentTestModels.hs
+++ b/persistent-test/src/PersistentTestModels.hs
@@ -11,6 +11,9 @@ import Init
 import PersistTestPetType
 import PersistTestPetCollarType
 
+-- just need to ensure this compiles
+import PersistentTestModelsImports()
+
 share [mkPersist persistSettings { mpsGeneric = True },  mkMigrate "testMigrate", mkDeleteCascade persistSettings, mkSave "_ignoredSave"] [persistUpperCase|
 
 -- Dedented comment

--- a/persistent-test/src/PersistentTestModelsImports.hs
+++ b/persistent-test/src/PersistentTestModelsImports.hs
@@ -1,0 +1,15 @@
+{-# language UndecidableInstances #-}
+
+-- | this just needs to compile
+module PersistentTestModelsImports where
+
+import Database.Persist.TH
+
+share [mkPersist sqlSettings] [persistUpperCase|
+
+User
+    name    String
+    age     Int
+    deriving Eq Show
+
+|]

